spec/support/shared_examples/sign_in.rb

```diff
@@ -324,16 +324,34 @@ def user_with_broken_personal_key(scenario)
     fake_analytics = FakeAnalytics.new
     allow_any_instance_of(ApplicationController).to receive(:analytics).
       and_wrap_original do |original|
-        original_analytics = original.call
-        if original_analytics.request.params[:controller] == 'users/sessions' &&
-           original_analytics.request.params[:action] == 'create'
-          expect(original_analytics.user).to eq(user)
+        if original.receiver.instance_of?(Users::SessionsController) &&
+           original.receiver.action_name == 'create'
+          expect(original.call.user).to eq(user)
           asserted_expected_user = true
         end
 
         fake_analytics
       end
 
+    fill_in_credentials_and_submit(user.email, 'wrongpassword')
+    expect(asserted_expected_user).to eq(true)
+    expect(fake_analytics).to have_logged_event(
+      'reCAPTCHA verify result received',
+      recaptcha_result: {
+        assessment_id: kind_of(String),
+        success: true,
+        score: 1.0,
+        errors: [],
+        reasons: [],
+      },
+      evaluated_as_valid: true,
+      score_threshold: 0.2,
+      recaptcha_action: 'sign_in',
+      form_class: 'RecaptchaMockForm',
+    )
+    asserted_expected_user = false
+    fake_analytics = FakeAnalytics.new
```

Comment:
would it be easier to add a `FakeAnalytics#reset!` or something instead of having to re-create an instance?

Reply:
> would it be easier to add a `FakeAnalytics#reset!` or something instead of having to re-create an instance?

Probably 👍 My only hesitation would be introducing a method to `FakeAnalytics` not present in `Analytics`, in case someone were to assume they shared the same API, but that feels pretty unlikely.

Reply:
Updated in c1117f2